src/ResourceManager/Sql/Commands.Sql/ManagedDatabase/Cmdlet/NewAzureSqlManagedDatabase.cs

```diff
@@ -0,0 +1,185 @@
+    public class NewAzureSqlManagedDatabase : AzureSqlManagedDatabaseCmdletBase<AzureSqlManagedDatabaseModel>
+    {
+        protected const string CreateNewByNameAndResourceGroupParameterSet =
+            "CreateNewManagedDatabaseFromInputParameters";
+
+        protected const string CreateNewByInputObjectParameterSet =
+            "CreateNewManagedDatabaseFromAzureSqlManagedInstanceModelInstanceDefinition";
+
+        /// <summary>
+        /// Gets or sets the name of the managed database to create.
+        /// </summary>
+        [Parameter(Mandatory = true,
+            Position = 0,
+            HelpMessage = "The name of the Azure SQL Managed Database to create.")]
+        [Alias("Name")]
+        [ValidateNotNullOrEmpty]
+        public string ManagedDatabaseName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the Azure Sql Managed instance to use
+        /// </summary>
+        [Parameter(ParameterSetName = CreateNewByNameAndResourceGroupParameterSet,
+            Mandatory = true,
+            Position = 1,
+            HelpMessage = "The Azure Sql Managed Instance name.")]
+        [ValidateNotNullOrEmpty]
+        public override string ManagedInstanceName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the resource group to use.
+        /// </summary>
+        [Parameter(ParameterSetName = CreateNewByNameAndResourceGroupParameterSet,
+            Mandatory = true,
+            Position = 2,
+            HelpMessage = "The name of the resource group.")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public override string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the Azure SQL Managed Database collation to use
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "The collation of the Azure SQL Managed Database collation to use.")]
+        [ValidateNotNullOrEmpty]
+        [PSArgumentCompleter("SQL_Latin1_General_CP1_CI_AS", "Latin1_General_100_CS_AS_SC")]
+        public string Collation { get; set; }
+
+        /// <summary>
+        /// Gets or sets the tags associated with the Azure Sql Managed Database
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "The tags to associate with the Azure Sql Managed Database")]
+        [Alias("Tag")]
+        public Hashtable Tags { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Azure Sql Managed Instance object
+        /// </summary>
+        [Parameter(ParameterSetName = CreateNewByInputObjectParameterSet,
+            Mandatory = true,
+            ValueFromPipeline = true,
+            HelpMessage = "The Azure Sql Managed Instance object")]
+        [ValidateNotNullOrEmpty]
+        [Alias("InputObject")]
+        public AzureSqlManagedInstanceModel ManagedInstance { get; set; }
```

Comment:
Should have a resourceId parameter set for this cmdlet that accepts a managedinstance resourceId (called ManagedInstanceResourceId).  This parameter should also be "ManagedInstanceObject"